# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/jit-test/tests/ion/bug1605641.js
# Commit: 6ccaefced3df
# Full Hash: 6ccaefced3df63b297057b31a856527153801a71
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2020-01-13 15:45:24
# Description:
#   Bug 1605641 - Mark 'done' stack value for destructuring assignment as having implicit uses. r=tcampbell
#   
#   We need to make sure this value isn't optimized out because the exception
#   handler may need it.
#   
# ==============================================================================

diff -r 96753184f3c4 -r 6ccaefced3df js/src/jit-test/tests/ion/bug1605641.js
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/js/src/jit-test/tests/ion/bug1605641.js	Mon Jan 13 09:23:17 2020 +0000
@@ -0,0 +1,8 @@
+// |jit-test| error:9
+function thrower() {
+    throw 9;
+}
+function f() {
+    return [...{} [thrower(...["foo"])]] = "undefined";
+}
+f();